### DISCUSSION TITLE
Changed C++ standard to C++17

### DISCUSSION THREAD

**Discussion by emjotde** [marian-nmt/marian-dev#920] [Announcements]
Hi all,
with the current sync we allow for C++17 now. Have fun.

**Comment by emjotde** [comment]
And the first piece of code that actually uses that is here:
https://github.com/marian-nmt/marian-dev/blob/b8bf086b109e92a4f304750a14cf99c1848b86a0/src/layers/guided_alignment.h#L50

**Comment by kpu** [comment]
intgemm got much easier to refactor.

**Comment by emjotde** [comment]
Gimme those sweet column-wise scalars :)
